views/hoc/AuthenticatedRoute.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useContext, useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { AuthContext } from '../contexts/AuthContext'
+import api from '../api'
+
+const AuthenticatedRoute = ({ children }) => {
+  const navigate = useNavigate()
+  const [user, setUser] = useContext(AuthContext)
+  const userId = window.sessionStorage.getItem('userId')
+
+  const fetchUser = async () => {
+    try {
+      const currentUser = await api.users.findOne(userId)
```

Comment:
what happens if the `userId` is changed? Can we do this in a way where we don't pass the user's ID and the back end uses the session data to authenticate and return the signed in user?

Reply:
[3eec4d0](https://github.com/AMP-SCZ/dpdash/pull/317/commits/3eec4d0ad5aa3f22f5514f03bedc26509b2f0c8e)